pkg/hns/endpoint_windows.go

```diff
@@ -40,6 +52,102 @@ func GetSandboxContainerID(containerID string, netNs string) string {
 	return containerID
 }
 
+func GenerateHnsEndpoint(epInfo *EndpointInfo, n *NetConf) (*hcsshim.HNSEndpoint, error) {
+	// run the IPAM plugin and get back the config to apply
+	hnsEndpoint, err := hcsshim.GetHNSEndpointByName(epInfo.EndpointName)
+	if err != nil && (err.Error() != fmt.Sprintf("Endpoint %s not found", epInfo.EndpointName)) {
+		return nil, errors.Annotatef(err, "Attempt to get endpoint \"%v\" failed", epInfo.EndpointName)
+	}
+
+	if hnsEndpoint != nil {
+		if hnsEndpoint.VirtualNetwork != epInfo.NetworkId {
+			_, err = hnsEndpoint.Delete()
+			if err != nil {
+				return nil, errors.Annotatef(err, "Failed to delete endpoint %v, err:%v", epInfo.EndpointName, err)
+			}
+			hnsEndpoint = nil
+		}
+	}
+
+	if hnsEndpoint == nil {
+		hnsEndpoint = &hcsshim.HNSEndpoint{
+			Name:           epInfo.EndpointName,
+			VirtualNetwork: epInfo.NetworkId,
+			DNSServerList:  strings.Join(epInfo.Nameservers, ","),
+			DNSSuffix:      strings.Join(epInfo.DnsSearch, ","),
+			GatewayAddress: epInfo.Gateway.String(),
+			IPAddress:      epInfo.IpAddress,
+			Policies:       n.MarshalPolicies(),
+		}
+	}
+	return hnsEndpoint, nil
+}
+
+func GenerateHcnEndpoint(epInfo *EndpointInfo, n *NetConf) (*hcn.HostComputeEndpoint, error) {
+	// run the IPAM plugin and get back the config to apply
+	hcnEndpoint, err := hcn.GetEndpointByName(epInfo.EndpointName)
+	if err != nil && (err != hcn.EndpointNotFoundError{EndpointName: epInfo.EndpointName}) {
```

Comment:
same here, `hcsshim.IsNotExist` is better than this struct comparison

Reply:
I'll have to make a change in hcsshim to do this one. I will fix in a later pr.